### PR TITLE
Weather tile inputs geolocation manually over HTTP, allows for location updating

### PR DESCRIPTION
**Pull request by matildepark**
Checks the webpage protocol. If it's HTTP, shows a new UI where you input your latitude and longitude manually, with a link to a site where you can easily get it. 

Strips the input of whitespace, tests the latlong against a regex and if it works, submits it to the back-end. Otherwise gives an error. Screenshots attached of the UI.

Screen Shot 2019-08-12 at 7 34 23 PM

Screen Shot 2019-08-12 at 7 55 07 PM

### CODE REVIEW

**Comment by ixv** [comment]
Remember to minify.

**Comment by matildepark** [comment]
Also adding in a quick line to let people update location. It uses the same functions as setting the location, including allowing the user to input manually over insecure contexts.

Screen Shot 2019-08-13 at 12 35 25 PM

**Comment by matildepark** [comment]
I know @Fang- has brought this up, but it may be preferable to refactor this PR into a UX flow that defaults to manual entry, with a "Detect location ->" option that lets the user opt-in (and ideally only shows the option over HTTPS to begin with).

This is a user flow question, so I'm not sure who takes ownership over that, but if you think it's better, it's not a big deal to amend it to work that way.

**Comment by loganallenc** [comment]
style nitpick: please put semicolons at the end of statements.

File: pkg/interface/weather/tile/tile.js [hunk 1]
```diff
@@ -58,6 +89,21 @@ export default class WeatherTile extends Component {
...
+    let error;
+    if (this.state.error === true) {
+     error = <p className="label-small red pt1">Incorrect latitude/longitude formatting. Please try again. <br/>(eg. "29.558107,-95.089023")</p>
+    }
+    return this.renderWrapper((
+      <div>
+        <a style={{"color": "white", "cursor": "pointer"}} onClick={() => this.setState({manualEntry: !this.state.manualEntry})}>&lt;&#45;</a>
+        <p className="label-regular white pt2">Please enter your <a href="https://latitudeandlongitude.org/" target="_blank">latitude and longitude</a>.</p>
+        {error}
+        <form className="flex absolute" style={{"bottom": "0"}}><input id="gps" className="white pb1 bg-transparent outline-0 bn bb-ns b--white" type="text" placeholder="29.558107,-95.089023" onKeyDown={this.keyPress.bind(this)}></input> <input className="bg-transparent inter white w-20 outliner-0 bn pointer" type="submit" onClick={() => this.manualLocationSubmit()} value="->"></input></form>
```
Comment (loganallenc) [inline pkg/interface/weather/tile/tile.js]: style nitpick: lines should not be greater than 80 characters

**Comment by matildepark** [comment]
Fixed style nitpicks and amended the UX to manual entry first. It shows 'detect location' if it's a secure context (HTTPS) and lets the user opt-in to geolocation. Also tried to make the input field look less weird.

Screen Shot 2019-08-13 at 8 36 22 PM

Screen Shot 2019-08-13 at 8 36 17 PM

**Pull request merged**